version: '3'

services:
  nginx:
    image: nginx:mainline-alpine3.18-perl
    container_name: nginx
    tty: true
    restart: always
    volumes:
    - ./conf/:/etc/nginx/conf.d/
    ports:
      - 8080:80
    networks:
      - appnode
  
  node:
    build:
      context: node
      dockerfile: Dockerfile.nodeapp
    container_name: node
    tty: true
    restart: always
    networks:
      - appnode
    depends_on:
      mysql:
        condition: service_healthy

  mysql: 
    image: mysql
    container_name: mysql
    tty: true
    command: --default-authentication-plugin=mysql_native_password
    environment:
      MYSQL_ROOT_PASSWORD: root
      MYSQL_DATABASE: nodeapp
    restart: always
    networks:
      - appnode
    healthcheck:
      test: [CMD,'mysqladmin','ping','-h','localhost','-u','root','-p$MYSQL_ROOT_PASSWORD']
      interval: 3s
      timeout: 14s
      retries: 5
    

networks:
  appnode:
    driver: bridge
